### ISSUE TITLE
Test: terminal editors should be marked as dirty when the shell has child processes

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#129428]
Refs: https://github.com/microsoft/vscode/issues/127193

- [ ] macOS 
- [ ] linux 
- [ ] windows 

Complexity: 2

Authors: @Tyriar, @meganrogge 

---
**Note that if you press `Don't Save`, the dirty indicator won't ever be shown again for that terminal.**

Check that:
- you don't see the indicator unless there's a process
- you see an indicator and get prompted when closing a terminal editor with a process

**Comment by sandy081** [comment]
I assume that this functionality is available only for terminals in editor.

**Comment by Tyriar** [comment]
@sandy081 yes, but you can turn it on elsewhere with the `terminal.integrated.confirmOnKill` setting. Updating the issue to include that.
